### PR TITLE
fix: the bug when htpasswd has multiple creds

### PR DESCRIPTION
**Pull request by tsnaik**
earlier, when you had more than one creds in htpasswd file separated by newline, it used to only read the first cred in the file and ignore the rest.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,4 +19,5 @@ test/data/
 .idea/
 coverage.html
 tags
-vendor/
```
Comment (rchincha) [inline .gitignore:22]: make this a separate commit

File: pkg/api/auth.go [hunk 2]
```diff
@@ -141,15 +141,16 @@ func basicAuthHandler(c *Controller) mux.MiddlewareFunc {
 			if err != nil {
 				panic(err)
 			}
+			defer f.Close()
 
-			for {
```
Comment (rchincha) [inline pkg/api/auth.go:145]: Can you add a test case to prove this path works?
1. Single entry with a newline at end of file
2. Single entry without a newline at end of file
3. Multiple entries with a newline at end of file
4. Multiple entries without a newline at end of file

Comment (tsnaik) [reply pkg/api/auth.go:145]: done

File: pkg/api/controller_test.go [hunk 3]
```diff
@@ -79,6 +93,131 @@ func TestNew(t *testing.T) {
...
+				So(resp.StatusCode(), ShouldEqual, 401)
+			}()
+		}
+	})
+}
+
+func TestHtpasswdTwoCreds(t *testing.T) {
+	Convey("Two creds", t, func() {
+		twoCredtests :=
+			[]string{"alice:$2y$12$wc6M7d7vTyFD3m9J2hTWPuM3q1gd1lM.8Epnpd8YheZCh6hx/vxma\n" +
```
Comment (rchincha) [inline pkg/api/controller_test.go:157]: Instead of static creds here, can we just something on the fly?
